### ISSUE TITLE
Allow more granular specification of proj crate features

### ISSUE THREAD

**Issue by urschrei** [georust/geo#439]
When the `proj-sys` and `proj` refactorings land (currently blocked by buggy upstream PROJ v7.0.0), allowing the use of `pkgconfig` (a common requirement on macOS) or including a bundled PROJ lib, we should extend the availability of these features to the `use-proj` feature. I don't know if you can specify features to "pass along", but if that isn't possible, we can add e.g.`use-proj-bundled` and `use-proj-pkgconfig` features, although that seems a bit clunky – I need to investigate.

**Comment by urschrei** [comment]
Closing in favour of #449, which I apparently created while in a fugue state.
